packages/data/README.md

```diff
@@ -72,21 +82,22 @@ registerStore( 'my-shop', {
 		},
 	},
 
+	controls: {
+		FETCH_FROM_API( action ) {
+			return apiFetch( { path: action.path } );
+		},
+	},
+
 	resolvers: {
-		async getPrice( state, item ) {
-			const price = await apiFetch( { path: '/wp/v2/prices/' + item } );
-			dispatch( 'my-shop' ).setPrice( item, price );
+		* getPrice( state, item ) {
+			const path = '/wp/v2/prices/' + item;
+			const price = yield actions.fetchFromAPI( path );
```

Comment:
Question: What do you think about the inconsistency of the actions behavior?

 - actions with controls returning something defined by the control
 - actions without controls returning the action itself or undefined?

Reply:
Are you suggesting we should promote `yield actions.setPrice( item, price )` for consistent way of causing the dispatch?

I'm not really sure there's an inconsistency here, in that the `actions.fetchFromAPI` itself is still just a plain action object; it's the act of yielding it which transforms its result into being assigned into the variable or return statement.

Reply:
I'm just thinking that maybe `fetchFromAPI` is not really an action and might be declared separately. `controls.fetchFromAPI`

Reply:
As I see it, the semantic purpose for an action is to express an intent. To that end, we traditionally consider its use in the context of dispatching within the store, but I don't see it as being fundamentally different from what we're proposing with controls, where the intent is processed as defined by the continuation procedure via the middleware.

It kinda speaks back to my thoughts at https://github.com/WordPress/gutenberg/pull/8096#issuecomment-409009965, where there are multiple things going on here: namely the handling of generator and the potentially-asynchronous continuation. I'm led to think that they are complementary for the purposes we're using them for, but also that it leads to open questions on: 

- Is it okay to have a generator action creator which doesn't cause any asynchronous continuation to occur?
- Is it okay for a control to be used without an attached asynchronous behavior (i.e. returns synchronously to assign / return on the `yield`)?

Both seem like implementation details that the action creator needn't be concerned with. It _could_ be asynchronous, or it could not. From the developer's perspective, it's important that it's consistent in how it's used: yielding can assign a value, whether that's assigned asynchronously or not. It's a nice bonus that it provides a solution for a common use-case (multi-dispatch).

Thinking on how this is at all different from effects, the one thing that stood out to me is that we quickly turned to effects as they were the only option to do either asynchronous or multi-dispatch for a while. And once something was converted to an effect, it became that much more convenient to stay in the effect handler to perform the myriad of behaviors associated with an action. By contrast, with the routines / controls, it establishes a simple and obvious pattern to temporarily escape out of the flow from within the action creator itself in an isolated fashion.